session_rpcserver.go

```diff
@@ -1224,10 +1187,15 @@ func (s *sessionRpcServer) AddAutopilotSession(ctx context.Context,
 			"autopilot server: %v", err)
 	}
 
-	// We only persist this session if we successfully retrieved the
-	// autopilot's static key.
-	sess.RemotePublicKey = remoteKey
-	if err := s.cfg.db.CreateSession(sess); err != nil {
+	err = s.cfg.db.UpdateSessionRemotePubKey(sess.LocalPublicKey, remoteKey)
+	if err != nil {
+		return nil, fmt.Errorf("error setting remote pubkey: %v", err)
+	}
+
+	// We only activate the session if the Autopilot server registration
+	// was successful.
+	sess, err = s.cfg.db.CreateSession(sess.ID)
+	if err != nil {
```

Comment:
There's nothing to change here, but just a reminder, as we discussed in the original session-linking PR, that we should allow session revocation based on `groupIDs` at the autopilot server level. 🤓 Otherwise, this could lead to a scenario where the autopilot server persists the reserved session, but `litd` removes it on the next startup if an error occurs here. That'd become problematic for linked sessions.

Reply:
ah yes, that's a great point

Reply:
opened an issue on the appropriate repo - thanks for the flag 🙏